dfsclient/src/main/java/org/hpccsystems/dfs/client/HpccRemoteFileReader.java

```diff
@@ -344,6 +335,35 @@ public HpccRemoteFileReader(FileReadContext ctx, DataPartition dp, IRecordBuilde
         openTimeMs = System.currentTimeMillis();
     }
 
+    private static Span createReadSpan(FileReadContext context, DataPartition dp)
+    {
+        String readSpanName = "HPCCRemoteFileReader/Read_" + dp.getFileName() + "_" + dp.getThisPart();
+        Span readSpan = Utils.createChildSpan(context.parentSpan, readSpanName);
+        readSpan.setStatus(StatusCode.OK);
+
+        String primaryIP = dp.getCopyIP(0);
+        String secondaryIP = "";
+        if (dp.getCopyCount() > 1)
+        {
+            secondaryIP = dp.getCopyIP(1);
+        }
+
+        long readSize = context.readSizeKB;
+        if (readSize < 0)
+        {
+            readSize = RowServiceInputStream.DEFAULT_MAX_READ_SIZE_KB;
+        }
+        readSize *= 1000;
+
+        Attributes attributes = Attributes.of(  AttributeKey.stringKey("server.0.address"), primaryIP,
+                                                AttributeKey.stringKey("server.1.address"), secondaryIP,
+                                                ServerAttributes.SERVER_PORT, Long.valueOf(dp.getPort()),
```

Comment:
minor (perhaps) but we've noticed APM/grafana tends to group numeric attributes seperately from string based attributes, forcing us to report ports as strings to keep them group with hostnames, etc.